packages/opentelemetry-resources/src/platform/node/ServiceInstanceIDDetector.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Detector } from '../../types';
+import { ResourceDetectionConfig } from '../../config';
+import { IResource } from '../../IResource';
+import { serviceInstanceIDDetectorSync } from '../browser/ServiceInstanceIDDetectorSync';
+
+/**
+ * ServiceInstanceIDDetector detects the resources related to the service instance ID.
+ */
+class ServiceInstanceIDDetector implements Detector {
+  detect(_config?: ResourceDetectionConfig): Promise<IResource> {
+    return Promise.resolve(serviceInstanceIDDetectorSync.detect());
+  }
+}
+
+export const serviceInstanceIDDetector = new ServiceInstanceIDDetector();
```

Comment:
We can remove this implmenetation and just implement `DetectorSync`. 

Some back-story on it:

We've deprecated `Detector` in favor of `DetectorSync` as people had to `await` resource detection even though most resources resolved synchronously. This was originally done for resources which have to be retrieved via an http call. To avoid users having to deal with `await`ing resources we introduced the concept of async-resolvable resource attributes (`IResource#waitForResourceAttributes()`) and moved the `await`ing of these to the SDKs export pipelines instead.

We kept the `Detector` implementations around because we could not break the interface. For new detectors we now recommend to only use `DetectorSync`